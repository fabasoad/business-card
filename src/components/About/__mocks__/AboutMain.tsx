import * as React from 'react'

export default function AboutMain() {
  return (
    <div data-testid="AboutMain"></div>
  )
}
