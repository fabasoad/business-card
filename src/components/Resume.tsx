/// <reference path='./Resume.d.ts' />
import * as React from 'react'
import { useTranslation } from 'react-i18next'

import pdfResume from '../assets/files/yf-cv.pdf'

export default function Resume() {
  const { t } = useTranslation()
  return (
    <div id="resume" className="light-component">
      <div className="section-title text-center center">
        <h2>{t('business-card-resume-title')}</h2>
        <hr />
      </div>
      <div className="d-flex justify-content-center">
        <a className="btn btn-primary btn-lg mr-1" href={pdfResume} rel="noopener noreferrer" target="_blank">
          <i className="fas fa-file-pdf"></i>
        </a>
        <a className="btn btn-primary btn-lg" href="https://drive.google.com/open?id=1RPKIdkOwcD_A4P0ZWQGrlFt5-abcN6-3CBJ7WvM8mc8" rel="noopener noreferrer" target="_blank">
          <i className="fas fa-file-word"></i>
        </a>
      </div>
    </div>
  )
}
