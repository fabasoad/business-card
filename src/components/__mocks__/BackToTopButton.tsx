import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
