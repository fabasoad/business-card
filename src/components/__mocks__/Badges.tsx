import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
