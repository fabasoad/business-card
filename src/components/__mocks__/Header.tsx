import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
