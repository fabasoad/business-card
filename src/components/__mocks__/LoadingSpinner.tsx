import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
