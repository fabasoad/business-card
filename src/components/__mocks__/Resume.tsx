import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
