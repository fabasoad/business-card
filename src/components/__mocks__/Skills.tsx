import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
