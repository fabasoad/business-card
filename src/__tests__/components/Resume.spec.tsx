/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Resume from '../../components/Resume'
import { useTranslation } from '../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render Resume correctly', () => {
  const wrapper: ShallowWrapper = shallow(<Resume />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(1)
  expect(tMock).toHaveBeenCalledWith('business-card-resume-title')
})
