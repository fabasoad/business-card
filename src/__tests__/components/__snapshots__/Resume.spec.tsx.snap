// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Resume correctly 1`] = `
<div
  className="light-component"
  id="resume"
>
  <div
    className="section-title text-center center"
  >
    <h2>
      business-card-resume-title
    </h2>
    <hr />
  </div>
  <div
    className="d-flex justify-content-center"
  >
    <a
      className="btn btn-primary btn-lg mr-1"
      href="test-file-stub"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fas fa-file-pdf"
      />
    </a>
    <a
      className="btn btn-primary btn-lg"
      href="https://drive.google.com/open?id=1RPKIdkOwcD_A4P0ZWQGrlFt5-abcN6-3CBJ7WvM8mc8"
      rel="noopener noreferrer"
      target="_blank"
    >
      <i
        className="fas fa-file-word"
      />
    </a>
  </div>
</div>
`;
